src/context/msal/constants.ts

```
@@ -0,0 +1,25 @@
+import { Configuration, PublicClientApplication } from "@azure/msal-browser";
+
+
+const clientId = process.env.REACT_APP_CLIENT_ID;
+const tenantId = process.env.REACT_APP_TENANT_ID;
+//For now, then we'll use a production url 
+const url = process.env.REACT_APP_DEV_URI;
+
+if(!clientId || !tenantId) {
+	throw new Error('Some of this values are undefined, please provide a correct value');
```

Might as well do 2 seperate checks and throw an error saying which variable is undefined and where the Dev can find it on Azure and where they should add it (ie. In the config file)

---

I did this like that since is not a good security practice to disclose which parameter is wrong in an authentication error, but you can tell me what do you think about it . However I can definitely do separate checks for this!

---

If the code runs in the front-end it won't matter since an attacker can just hair execution and try himself. At best that would be classified as "security through obfuscation"...   In other words you're fine to say which variable is missing.

However, we should include build or deploy checks which fail if all relevant configuration is not provided and at that point it's for the dev's eyes only... That will have to be a separate ticket however so leave that for now.